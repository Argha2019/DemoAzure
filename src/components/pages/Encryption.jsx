import CryptoJS from 'crypto-js';

/**
 * @author utssaha(Utshab Saha)
 * 
 * This file is responsible or our own encryption and decryption. We can implement any number of 
 * algorithm and use it in other classes
 */
const secretKey = 'theNightWalkers';
const salt = "3FF2EC019C627B945225DEBAD71A01B6985FE84C95A70EB132882F88C0A59A55";
const iv = "12345678901234567890123456789012";

/**
 * This function uses AES algorithm for generating the key
 * @returns String
 */
const generateKey = () => {
    return CryptoJS.PBKDF2(secretKey, CryptoJS.enc.Hex.parse(salt), {
        keySize: 128/32,
        iterations: 10
    });
};

/**
 * This function uses AES algorithm for encryption
 * @param data 
 * @type String
 * @returns String
 */
const encryptByAES = (data) => {
    let key = generateKey();
    return CryptoJS.AES.encrypt(data, key, {
        iv: CryptoJS.enc.Hex.parse(iv)
    }).ciphertext.toString(CryptoJS.enc.Base64);
};

/**
 * This function uses AES algorithm for decryption
 * @param data
 * @type String
 * @returns String
 */
const decryptByAES = (data) => {
    let key = generateKey();
    let cipherParams = CryptoJS.lib.CipherParams.create({
        ciphertext: CryptoJS.enc.Base64.parse(data)
    });
    return CryptoJS.AES.decrypt(cipherParams, key, {
        iv: CryptoJS.enc.Hex.parse(iv)
    }).toString(CryptoJS.enc.Utf8);
};

export { encryptByAES, decryptByAES };
