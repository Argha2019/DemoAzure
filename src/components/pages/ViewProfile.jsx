import React from 'react';
import Cookies from 'js-cookie';
import ViewTemplate from "../templates/ViewTemplate/Template.jsx";
import { Row, Col, Form } from 'react-bootstrap';
import Image from 'react-bootstrap/Image';
import { decryptByAES } from './Encryption.jsx';

/**
 * @author Arghadeep Mandal
 * This file is responsible for making the header and footer template for role:"VIEW"
 * which is always rendered in every page. 
 */


export default class ViewProfile extends React.Component {
	constructor(props) {
		super(props);
		let profileData = {};
		try {
			profileData = JSON.parse(decryptByAES(Cookies.get('viewData')))[0];
		}
		catch (e) {
			profileData = {
				firstName: '',
				lastName: '',
				username: '',
				role: '',
				phone: '',
				mail: '',
				addressLine1: '',
				addressLine2: '',
				state: '',
				country: '',
				pincode: '',
				dateOfBirth: ''
			};
		}
		this.state = {
			firstName: profileData.firstName,
			lastName: profileData.lastName,
			dateOfBirth: profileData.dob,
			phone: decryptByAES(profileData.phone),
			mail: decryptByAES(profileData.mail),
			addressLine1: profileData.addressLine1,
			addressLine2: profileData.addressLine2,
			state: profileData.state,
			country: profileData.country,
			pincode: profileData.pinCode,
			role: profileData.roles,
			username: decryptByAES(profileData.username),
		};
	}




	render() {
		return (
			<ViewTemplate>
				<div className="register-wrap" style={{ height: "1000px" }}>
				<br />
					<br />
					<br />
					<br />
					<h1 className="fontWhite">Profile</h1>
					<div className="Oms register-html">
					<Form className="MyForm register-form">
						<Form.Group as={Row} controlId="username">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">Username</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control name="username" type="text" defaultValue={this.state.username} ref="username" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="role">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">Role</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control name="role" type="text" defaultValue={this.state.role} ref="role" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="firstName">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">First Name</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control className="firstName" name="firstName" type="text" defaultValue={this.state.firstName} ref="firstName" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="lastName">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">Last Name</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control className="lastName" name="lastName" type="text" defaultValue={this.state.lastName} ref="lastName" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="dateOfBirth">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">Date of Birth</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control name="dateOfBirth" type="date" defaultValue={this.state.dateOfBirth} ref="dateOfBirth" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="phone">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">Phone</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control className="phone" name="phone" type="number" defaultValue={this.state.phone} ref="phone" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="mail">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">Mail ID</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control className="mail" name="mail" type="email" defaultValue={this.state.mail} ref="mail" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="addressLine1">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">Address Line 1</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control className="addressLine1" name="addressLine1" type="text" defaultValue={this.state.addressLine1} ref="addressLine1" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="addressLine2">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">Address Line 2</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control className="addressLine2" name="addressLine2" type="text" defaultValue={this.state.addressLine2} ref="addressLine2" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="state">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">State</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control className="state" name="state" type="text" defaultValue={this.state.state} ref="state" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="country">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">Country</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control className="country" name="country" type="text" defaultValue={this.state.country} ref="country" readOnly required />
							</Col>
						</Form.Group>

						<Form.Group as={Row} controlId="pincode">
							<Col sm={4} md={3} className="alignLabel">
								<Form.Label className="label Labelling">Pincode</Form.Label>
							</Col>
							<Col sm={8} md={9}>
								<Form.Control className="pincode" name="pincode" type="number" defaultValue={this.state.pincode} ref="pincode" readOnly required />
							</Col>
						</Form.Group>
					</Form>
				</div>
				</div>
			</ViewTemplate>
		);
	}
}
