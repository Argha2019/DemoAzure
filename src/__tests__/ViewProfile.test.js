import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ViewProfile from '../components/pages/ViewProfile.jsx';
import Cookies from 'js-cookie';
import { encryptByAES, decryptByAES } from '../components/pages/Encryption.jsx';

const dummyData = {
    firstName: "test",
    lastName: "test",
    dateOfBirth: "yyyy-mm-dd",
    phone: encryptByAES("1234567890"),
    mail: encryptByAES("test@test.com"),
    addressLine1: "test",
    addressLine2: "test",
    state: "test",
    country: "test",
    pincode: "test",
    role: "AGENT VIEW",
    username: encryptByAES("test123")
};

describe('ViewProfile', () => {
    it('render component', () => {
        Cookies.set('viewData', encryptByAES(JSON.stringify([dummyData])));
        const wrapper = shallow(<ViewProfile />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
})