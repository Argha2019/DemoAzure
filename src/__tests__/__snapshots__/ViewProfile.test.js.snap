// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewProfile render component 1`] = `
<ViewTemplate>
  <div
    className="register-wrap"
    style={
      Object {
        "height": "1000px",
      }
    }
  >
    <br />
    <br />
    <br />
    <br />
    <h1
      className="fontWhite"
    >
      Profile
    </h1>
    <div
      className="Oms register-html"
    >
      <Form
        className="MyForm register-form"
        inline={false}
      >
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="username"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Username
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              defaultValue="test123"
              name="username"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="role"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Role
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              name="role"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="firstName"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              First Name
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="firstName"
              defaultValue="test"
              name="firstName"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="lastName"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Last Name
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="lastName"
              defaultValue="test"
              name="lastName"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="dateOfBirth"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Date of Birth
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              name="dateOfBirth"
              readOnly={true}
              required={true}
              type="date"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="phone"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Phone
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="phone"
              defaultValue="1234567890"
              name="phone"
              readOnly={true}
              required={true}
              type="number"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="mail"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Mail ID
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="mail"
              defaultValue="test@test.com"
              name="mail"
              readOnly={true}
              required={true}
              type="email"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="addressLine1"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Address Line 1
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="addressLine1"
              defaultValue="test"
              name="addressLine1"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="addressLine2"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Address Line 2
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="addressLine2"
              defaultValue="test"
              name="addressLine2"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="state"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              State
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="state"
              defaultValue="test"
              name="state"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="country"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Country
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="country"
              defaultValue="test"
              name="country"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="pincode"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="label Labelling"
              column={false}
              srOnly={false}
            >
              Pincode
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="pincode"
              name="pincode"
              readOnly={true}
              required={true}
              type="number"
            />
          </Col>
        </FormGroup>
      </Form>
    </div>
  </div>
</ViewTemplate>
`;
